packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -153,11 +151,13 @@ class DataSourceWithBackend<
   /**
    * Override to skip executing a query
    *
-   * @returns false if the query should be skipped
+   * @returns false if the query should be skipped, override to customize logic
    *
    * @virtual
    */
-  filterQuery?(query: TQuery): boolean;
+  filterQuery(query: TQuery): boolean {
+    return !query.hide;
```

Comment:
We have gone back and forth on this a couple times.  We had this before, but it means that backend datasources can not use "hidden" as a way to have values in other queries: datadog, backend expressions come to mind but I know there are more.

For expressions we absolutely can not put this in the frontend, but perhaps the backend query handler can check if the datasource explicitly wants hidden queries and filter there?

Reply:
I added this to ExpressionDataSource 
`ts
filterQuery() {    return true;  }
`

Reply:
Maybe here:
https://github.com/grafana/grafana/blob/main/pkg/api/metrics.go#L44

Reply:
@briangann -- do you remember where else hidden queries are used in the backend?  for sure datadog